src/main/java/org/jabref/gui/importer/ImportEntriesDialog.java

```diff
@@ -163,13 +180,50 @@ private void initialize() {
 
                     return container;
                 })
-                .withOnMouseClickedEvent((entry, event) -> entriesListView.getCheckModel().toggleCheckState(entry))
+                .withOnMouseClickedEvent((entry, event) -> {
+                    entriesListView.getCheckModel().toggleCheckState(entry);
+                    displayBibTeX(entry);
+                })
                 .withPseudoClass(entrySelected, entriesListView::getItemBooleanProperty)
                 .install(entriesListView);
 
         selectedItems.textProperty().bind(Bindings.size(entriesListView.getCheckModel().getCheckedItems()).asString());
         totalItems.textProperty().bind(Bindings.size(entriesListView.getItems()).asString());
         entriesListView.setSelectionModel(new NoSelectionModel<>());
+        initBibTeX();
+    }
+
+    private void displayBibTeX(BibEntry entry) {
+        if (entriesListView.getCheckModel().isChecked(entry)) {
+            bibTeXData.clear();
+            try {
+                bibTeXData.appendText(getSourceString(entry));
+                bibTeXData.moveTo(0);
+                bibTeXData.requestFollowCaret();
+            } catch (
+                    IOException aE) {
+                bibTeXData.clear();
+            }
+        } else {
+            bibTeXData.clear();
+        }
+    }
+
+    private void initBibTeX() {
+        bibTeXData.setBorder(new Border(new BorderStroke(Color.GREY, BorderStrokeStyle.SOLID, CornerRadii.EMPTY, BorderWidths.DEFAULT)));
+        bibTeXData.setPadding(new Insets(5.0));
+        showEntryInformation.selectedProperty().addListener((observableValue, old_val, new_val) -> {
+            bibTeXDataBox.setVisible(new_val);
+            bibTeXDataBox.setManaged(new_val);
+        });
+    }
+
+    private String getSourceString(BibEntry entry) throws IOException {
```

Comment:
Can you move the whole logic please to the viewModel?

Reply:
Done